app/controllers/concerns/foreman/controller/registration.rb

```diff
@@ -70,4 +73,21 @@ def not_found(options = nil)
 
     false
   end
+
+  def registration_url
+    uri = if params[:url].present?
+            URI.join(params[:url], '/register')
```

Comment:
You're not likely to hit this, but this does ensure that the Smart Proxy must be mounted on `/`:
```
[1] pry(main)> require 'uri'
=> false
[2] pry(main)> URI.join('https://example.com/bla', '/register')
=> #<URI::HTTPS https://example.com/register>
```

Is it intention that you ignore the path provided in the URL? On the plus side, it also throws away any GET parameters so it's guaranteed that a string concat later succeeds.

Reply:
> Is it intention that you ignore the path provided in the URL

Yes, it is. I wanted to be 100% sure that url will be `.../register`, never came up to my mind that proxy can be mounted somewhere else than `/`. Not sure what to do now.

- If I won't reset path, user can by accident pass URL with path, which will cause errors later in the template.
- But if I will keep the behavior as it is now, URL from proxy mounted somewhere else than `/` simply won't work.

Reply:
I think it's ok to accept it for now. The Proxy isn't mounted elsewhere than `/` in practice so I think it's safe to assume a lot of other things would break if you tried to do so.